src/datafactory/azext_datafactory/generated/custom.py

```diff
@@ -0,0 +1,728 @@
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
+# pylint: disable=too-many-lines
+# pylint: disable=unused-argument
+
+import json
+from knack.util import CLIError
+from azure.cli.core.util import sdk_no_wait
+
+
+def datafactory_factory_list(client,
+                             resource_group_name=None):
+    if resource_group_name:
+        return client.list_by_resource_group(resource_group_name=resource_group_name)
+    return client.list()
+
+
+def datafactory_factory_show(client,
+                             resource_group_name,
+                             factory_name,
+                             if_none_match=None):
+    return client.get(resource_group_name=resource_group_name,
+                      factory_name=factory_name,
+                      if_none_match=if_none_match)
+
+
+def datafactory_factory_create(client,
+                               resource_group_name,
+                               factory_name,
+                               if_match=None,
+                               location=None,
+                               tags=None,
+                               factory_vsts_configuration=None,
+                               factory_git_hub_configuration=None,
+                               global_parameters=None):
+    all_repo_configuration = []
+    if factory_vsts_configuration is not None:
+        all_repo_configuration.append(factory_vsts_configuration)
+    if factory_git_hub_configuration is not None:
+        all_repo_configuration.append(factory_git_hub_configuration)
+    if len(all_repo_configuration) > 1:
+        raise CLIError('at most one of  factory_vsts_configuration, factory_git_hub_configuration is needed for '
+                       'repo_configuration!')
+    repo_configuration = all_repo_configuration[0] if len(all_repo_configuration) == 1 else None
+    return client.create_or_update(resource_group_name=resource_group_name,
+                                   factory_name=factory_name,
+                                   if_match=if_match,
+                                   location=location,
+                                   tags=tags,
+                                   identity=json.loads("{\"type\": \"SystemAssigned\"}"),
+                                   repo_configuration=repo_configuration)
+
+
+def datafactory_factory_update(client,
+                               resource_group_name,
+                               factory_name,
+                               tags=None):
+    return client.update(resource_group_name=resource_group_name,
+                         factory_name=factory_name,
+                         tags=tags,
+                         identity=json.loads("{\"type\": \"SystemAssigned\"}"))
```

Comment:
Would this also work?
```suggestion
                         identity={"type": "SystemAssigned"})
```

Reply:
let me check

Reply:
But this is not generic update

Reply:
This should works. https://github.com/Azure/azure-cli-extensions/pull/1321/files/0d11d629783241c0ceff9fcb8810ac18ec96a9e9?file-filters%5B%5D=.py#diff-33d334bab3112a59463530ec4bf8a302R333-R339. Also the test is working  https://github.com/Azure/azure-cli-extensions/pull/1321/files/0d11d629783241c0ceff9fcb8810ac18ec96a9e9?file-filters%5B%5D=.yaml#diff-f74a8cf39a468a7785c7db13716c4ed3R143-R167

Reply:
Got your point now.

Reply:
Changed. thx